server/cmd/reindex-search.go

```diff
@@ -0,0 +1,441 @@
+		_, err = meilisearchx.Client.Documents("dega").DeleteAllDocuments()
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		if err = AddPosts(false); err != nil {
+			log.Fatal(err)
+		}
+
+		if err = AddPosts(true); err != nil {
+			log.Fatal(err)
+		}
+
+		if err = AddCategories(); err != nil {
+			log.Fatal(err)
+		}
+
+		if err = AddTags(); err != nil {
+			log.Fatal(err)
+		}
+
+		if err = AddMedium(); err != nil {
+			log.Fatal(err)
+		}
+
+		if err = AddMenu(); err != nil {
+			log.Fatal(err)
+		}
+
+		if err = AddSpace(); err != nil {
+			log.Fatal(err)
+		}
+
+		if err = AddClaim(); err != nil {
+			log.Fatal(err)
+		}
+
+		if err = AddClaimant(); err != nil {
+			log.Fatal(err)
+		}
+
+		if err = AddRating(); err != nil {
+			log.Fatal(err)
+		}
+
+		if err = AddPodcast(); err != nil {
+			log.Fatal(err)
+		}
+
+		if err = AddEpisode(); err != nil {
+			log.Fatal(err)
+		}
+	},
+}
+
+func AddPosts(page bool) error {
+	posts := make([]model.Post, 0)
+	config.DB.Model(&model.Post{}).Where("is_page = ?", page).Preload("Format").Preload("Tags").Preload("Categories").Find(&posts)
+
+	postAuthorMap := make(map[uint][]uint)
+	postAuthors := make([]model.PostAuthor, 0)
+	config.DB.Model(&model.PostAuthor{}).Find(&postAuthors)
+	for _, pa := range postAuthors {
+		postAuthorMap[pa.PostID] = append(postAuthorMap[pa.PostID], pa.AuthorID)
+	}
+
+	postClaimsMap := make(map[uint][]uint)
+	postClaims := make([]factCheckModel.PostClaim, 0)
+	config.DB.Model(&factCheckModel.PostClaim{}).Find(&postClaims)
+	for _, pc := range postClaims {
+		postClaimsMap[pc.PostID] = append(postClaimsMap[pc.PostID], pc.ClaimID)
+	}
+
+	meiliPostObjects := make([]map[string]interface{}, 0)
+	for _, p := range posts {
+		var meiliPublishDate int64
+		if p.Status == "publish" {
+			meiliPublishDate = p.PublishedDate.Unix()
+		}
+
+		tagIDs := make([]uint, 0)
+		categoryIDs := make([]uint, 0)
+
+		for _, each := range p.Categories {
+			categoryIDs = append(categoryIDs, each.ID)
+		}
+		for _, each := range p.Tags {
+			tagIDs = append(tagIDs, each.ID)
+		}
+
+		meiliObj := map[string]interface{}{
+			"object_id":      fmt.Sprint("post_", p.ID),
+			"id":             p.ID,
+			"kind":           "post",
+			"title":          p.Title,
+			"subtitle":       p.Subtitle,
+			"slug":           p.Slug,
+			"status":         p.Status,
+			"excerpt":        p.Excerpt,
+			"description":    p.Description,
+			"is_featured":    p.IsFeatured,
+			"is_sticky":      p.IsSticky,
+			"is_highlighted": p.IsHighlighted,
+			"is_page":        p.IsPage,
+			"format_id":      p.FormatID,
+			"published_date": meiliPublishDate,
+			"meta_fields":    p.MetaFields,
+			"space_id":       p.SpaceID,
+			"tag_ids":        tagIDs,
+			"category_ids":   categoryIDs,
+			"author_ids":     postAuthorMap[p.ID],
+		}
+
+		if page {
+			meiliObj["object_id"] = fmt.Sprint("page_", p.ID)
+			meiliObj["kind"] = "page"
+		}
+
+		if p.Format.Slug == "fact-check" {
+			meiliObj["claim_ids"] = postClaimsMap[p.ID]
+		}
```

Comment:
We can fetch all posts at once. Then change object_id and kind based `p.is_page`. Why separate `page` variable?

Reply:
Yaa sure we can do that.. I'll make that change